### PR TITLE
Milestone 2: To-Do List app - Add & remove functionality

### PR DESCRIPTION
**Pull request by binyamolango**
# Hi Reviewer 👋

> Here are the changes I implemented

- Removed all hardcoded items from the tasks array.
- Created a new JavaScript file for the new functionality.
- Implemented a function for adding a new task (add a new element to the array).
- Implemented a function for deleting a task (remove an element from the array).
- Implemented a function for editing task descriptions.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,39 +1,24 @@
...
-    index: 3,
-    description: 'Coding challenge',
-    completed: false,
-  },
-  {
-    index: 4,
-    description: 'TAWG',
-    completed: true,
-  },
-];
```
Comment (laiifuu) [inline src/index.js:25]: Nice job removing the hardcoded array items, however the array itself should remain and functionality of adding and removing to do items should be done using this array.

File: src/add-rmv-edit.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  taskIn.value = '';
+  if (tasksItem === null) return;
+  const task = {
+    description: tasksItem,
+    completed: false,
+    index: tasks.length,
+  };
+  const filtered = [...tasks, task];
+  localStorage.setItem('listItem', JSON.stringify(filtered));
+};
```
Comment (laiifuu) [inline src/add-rmv-edit.js:35]: As mentioned in the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20adding%20a%20new%20task%20(add%20a%20new%20element%20to%20the%20array).), adding an item should be in its own function, the item should be added to an array that's declared (check the previous comment) and it should be separate from saving to the local storage. After an item is added to that array, then you can go ahead and save that array to the local storage.

File: src/add-rmv-edit.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const clicked = e.target.closest('.move');
+  if (!clicked) return;
+  const tasks = JSON.parse(localStorage.getItem('listItem')) || [];
+  const listNum = +clicked.dataset.remove;
+  const filtered = tasks.filter((task) => task.index !== listNum);
+  let filterOrder = [];
+  filtered.forEach((task, count) => {
+    task.index = count;
+    filterOrder = [...filterOrder, task];
+  });
```
Comment (laiifuu) [inline src/add-rmv-edit.js:53]: As mentioned in the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20deleting%20a%20task%20(remove%20an%20element%20from%20the%20array).), removing an item from a list should be in its own function. It should manipulate the declared to do list array.

File: src/add-rmv-edit.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+});
+
+todoList.addEventListener('click', (e) => {
+  const clicked = e.target.closest('.list');
+  if (!clicked) return;
+  clicked.addEventListener('keyup', () => {
+    const tasks = JSON.parse(localStorage.getItem('listItem')) || [];
+    const listNum = +clicked.dataset.desc;
+    const task = tasks.find((task) => task.index === listNum);
+    task.description = clicked.value;
```
Comment (laiifuu) [inline src/add-rmv-edit.js:65]: As mentioned in the [requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md#:~:text=Implement%20a%20function%20for%20editing%20task%20descriptions.), editing a task should be done in its own function, and it should manipulate the declared to do list array.

**Review by laiifuu — CHANGES_REQUESTED** [review]
## Hi @binyamolango  :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
cat-driving-serious

## Highlights
- [x] Nice UI :+1: 
- [x] Green linters check :heavy_check_mark: 
- [x] Descriptive PR :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @binyamolango, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Correct **Semantics** used ✅ 
- Well-structured project ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**